### DISCUSSION TITLE
Rethink just a single toml file

### DISCUSSION THREAD

**Discussion by hammerandtongs** [flo-at/minmon#20] [General]
The reasoning behind many/most monitoring solutions having "fancy" config directory structure is that its a requirement.

People need to be able to put monitoring in place for something they bring up with automated tools.

Editing a toml file is a nonstarter (and no concating it in doesn't work either).

I would recommend going through the mental or actual code exercise of parsing some example nagios configs. Steelman why things are the way they are and you'll find it illuminating (I did it a decade ago for an in house golang monitoring server). 

https://en.wikipedia.org/wiki/Straw_man#Steelmanning

**Comment by flo-at** [comment]
I'll take a look at it, thanks! It would be possible to either pass a single file or a directory and treat both cases differently. I'm not sure if automated deployments are a good use case for this tool though. It's targeting single-machine setups, not a client-server model which is typically used to gather the data from multiple machines.

**Comment by flo-at** [comment]
>and no concating it in doesn't work either

Can you give an example why?\
I get why multiple config file support has some advantages but I don't find an example that wouldn't work the same way with a concatenated single file. Is it just that the usual deployment tools won't handle the concatenation or is there a more complex reason I'm overlooking?

I'm asking because, if I were to implement this, I would try to concatenate the files internally. This way the TOML file parser doesn't need to be changed.

**Comment by hammerandtongs** [comment]
I'm adding devices to the kitchen or the living room (home automation on the mind atm).

If I cat those devices into a single file, pretty soon my tree is disordered.

I don't really assume I know what people would use to deploy their services or devices, but a print template  or file copy is usually within most peoples toolkit.

Inserting into a tree is easy with a directory layout but hard with a single file or directory.

**Reply by flo-at** [reply]
Fair enough. Shouldn't be too hard to implement. I'll move it to the issues again.

**Comment by flo-at** [comment]
I'll lock the discussion in favor to the ticket. It's on the roadmap for v1.0.
